### PR TITLE
[17.0][MIG] product_manufacturer: Migration to 17.0

### PR DESCRIPTION
**Pull request by NICO-SOLUTIONS**
Standard migration to 17.0

### CODE REVIEW

**Review by smcqueen81 — APPROVED** [review]
Funcional ok

**Review by HviorForgeFlow — APPROVED** [review]
Code review

File: product_manufacturer/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            lambda template: len(template.product_variant_ids) == 1
+        )
+        for template in unique_variants:
+            template.manufacturer_id = template.product_variant_ids.manufacturer_id
+            template.manufacturer_pname = (
+                template.product_variant_ids.manufacturer_pname
+            )
+            template.manufacturer_pref = template.product_variant_ids.manufacturer_pref
+            template.manufacturer_purl = template.product_variant_ids.manufacturer_purl
+        for template in self - unique_variants:
```
Comment (rousseldenis) [inline product_manufacturer/models/product_template.py]:
```suggestion
        (self - unique_variants).update({
            "manufacturer_id": False,
            "manufacturer_pname": False,
            "manufacturer_pref": False,
            "manufacturer_purl": False
        })
```

File: product_manufacturer/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        "product_variant_ids.manufacturer_id",
+        "product_variant_ids.manufacturer_pname",
+        "product_variant_ids.manufacturer_pref",
+        "product_variant_ids.manufacturer_purl",
+    )
+    def _compute_manufacturer_info(self):
+        unique_variants = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1
+        )
+        for template in unique_variants:
```
Comment (rousseldenis) [inline product_manufacturer/models/product_template.py:46]:
```suggestion
        unique_variants.update({
            "manufacturer_id": (...),
            "manufacturer_pname": (...),
            "manufacturer_pref": (...),
            "manufacturer_purl": (...),
        })
```

File: product_manufacturer/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        for template in self - unique_variants:
+            template.manufacturer_id = False
+            template.manufacturer_pname = False
+            template.manufacturer_pref = False
+            template.manufacturer_purl = False
+
+    def _inverse_manufacturer_info(self):
+        for template in self:
+            if len(template.product_variant_ids) == 1:
+                template.product_variant_ids.manufacturer_id = template.manufacturer_id
```
Comment (rousseldenis) [inline product_manufacturer/models/product_template.py]: Same as above

File: product_manufacturer/tests/test_product_manufacturer.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+
+from odoo.tests.common import TransactionCase
+
+
+class TestProductManufacturer(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```
Comment (rousseldenis) [inline product_manufacturer/tests/test_product_manufacturer.py]: FYI, Odoo has done it in a common test class now: https://github.com/odoo/odoo/blob/17.0/odoo/addons/base/tests/common.py#L21

**Review by rousseldenis — COMMENTED** [review]
Code review

**Comment by bizzappdev** [comment]
@NICO-SOLUTIONS I have made the review changes https://github.com/NICO-SOLUTIONS/product-attribute/pull/1 to fast-track the process. Please check and proceed with the merge if it feels good.

**Comment by HviorForgeFlow** [comment]
@rousseldenis all comments attended

**Review by CasVissers-360ERP — APPROVED** [review]
Functional review

**Review by antoniocanovas — APPROVED** [review]
Functional review OK !!

**Comment by Crazor** [comment]
This one also appears to be stuck.

File: product_manufacturer/tests/test_product_manufacturer.py [hunk 5]
```diff
@@ -8,7 +8,6 @@ class TestProductManufacturer(TransactionCase):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
-        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```
Comment (rousseldenis) [inline product_manufacturer/tests/test_product_manufacturer.py]: Either you let this, either you inherit from BaseCommon https://github.com/odoo/odoo/blob/17.0/odoo/addons/base/tests/common.py#L21

This allows to enhance tests performances.

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by aessig** [comment]
Looking forward for it to be merged

**Comment by NICO-SOLUTIONS** [comment]
@StefanRijnhart thx

**Pull request merged**